engine/src/main/java/org/terasology/engine/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -129,16 +129,16 @@ public void registerEventHandler(ComponentSystem handler) {
 
                 logger.debug("Found method: {}", method);
                 if (!Event.class.isAssignableFrom(types[0]) || !EntityRef.class.isAssignableFrom(types[1])) {
-                    logger.error("Invalid event handler method: {}", method.getName());
+                    logger.atError().addArgument(() -> method.getName()).log("Invalid event handler method: {}");
                     return;
                 }
 
                 requiredComponents.addAll(Arrays.asList(receiveEventAnnotation.components()));
                 List<Class<? extends Component>> componentParams = Lists.newArrayList();
                 for (int i = 2; i < types.length; ++i) {
                     if (!Component.class.isAssignableFrom(types[i])) {
-                        logger.error("Invalid event handler method: {} - {} is not a component class",
-                                method.getName(), types[i]);
+                        logger.atError().addArgument(() -> method.getName()).addArgument(types[i]).
+                                log("Invalid event handler method: {} - {} is not a component class");
```

Comment:
O(1) method call, although potentially performed once per parameter of each event handler method, it's only called in unexpected cases so low impact and error-level log so unlikely that call is done "in vain"
```suggestion
                        logger.error("Invalid event handler method: {} - {} is not a component class",
                                method.getName(), types[i]); //NOPMD
```